cli/cmd/compose/up.go

```diff
@@ -284,10 +289,18 @@ func setup(ctx context.Context, opts composeOptions, services []string) (*client
 		project.Services[0].DomainName = opts.DomainName
 	}
 	if opts.Build {
-		for _, service := range project.Services {
+		for i, service := range project.Services {
 			service.PullPolicy = types.PullPolicyBuild
+			project.Services[i] = service
```

Comment:
did we have a bug here not reaffecting modified service ?

Reply:
seems so